### ISSUE TITLE
iOS Build fail in Script "[CP] Embed Pods Frameworks" with invalid path 

### ISSUE THREAD

**Issue by virtualmarc** [flutter/flutter#123852]
## Steps to Reproduce

1. Execute `flutter build ios`
2. Switch to ios subdirectory
3. Execute `pub install`
4. Run fastlane target containing:
```
build_app(
  scheme: "Runner",
  archive_path: "./build/Runner.xcarchive",
  export_method: "app-store",
  output_directory: "./build/Runner"
)
```

**Expected results:** Build successful

**Actual results:** Build failed

## Environment

* Mac Mini M1 2020 with macOS Ventura 13.0.1
* Xcode 14.3 (14E222b)
* Flutter 3.7.9

```
Flutter 3.7.9 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 62bd79521d (16 hours ago) • 2023-03-30 10:59:36 -0700
Engine • revision ec975089ac
Tools • Dart 2.19.6 • DevTools 2.20.1
```

## Build Logs

<details>
  <summary>Build Output</summary>

```
▸ Processing Info.plist
▸ Linking Runner
▸ Generating 'Runner.app.dSYM'
▸ Running script 'Thin Binary'
▸ Running script 'Crashlytics'
▸ Running script '[CP] Embed Pods Frameworks'
** ARCHIVE FAILED **


The following build commands failed:
	PhaseScriptExecution [CP]\ Embed\ Pods\ Frameworks /Users/jenkins/Library/Developer/Xcode/DerivedData/Runner-fsgvokzjdkqkrmbocrmrxbuzfayy/Build/Intermediates.noindex/ArchiveIntermediates/Runner/IntermediateBuildFilesPath/Runner.build/Release-iphoneos/Runner.build/Script-5B61BB44B89A7C0231C51373.sh (in target 'Runner' from project 'Runner')
(1 failure)
[11:36:34]: Exit status: 65

+---------------+-------------------------+
|            Build environment            |
+---------------+-------------------------+
| xcode_path    | /Applications/Xcode.app |
| gym_version   | 2.212.1                 |
| export_method | app-store               |
| sdk           | iPhoneOS16.4.sdk        |
+---------------+-------------------------+

[11:36:34]: ▸ sent 29 bytes  received 20 bytes  98.00 bytes/sec
[11:36:34]: ▸ total size is 0  speedup is 0.00
[11:36:34]: ▸ rsync error: some files could not be transferred (code 23) at /AppleInternal/Library/BuildRoots/810eba08-405a-11ed-86e9-6af958a02716/Library/Caches/com.apple.xbs/Sources/rsync/rsync/main.c(996) [sender=2.6.9]
[11:36:34]: ▸ Command PhaseScriptExecution failed with a nonzero exit code
[11:36:34]: 
[11:36:34]: ⬆️  Check out the few lines of raw `xcodebuild` output above for potential hints on how to solve this error
[11:36:34]: 📋  For the complete and more detailed error log, check the full log at:
[11:36:34]: 📋  /Users/jenkins/Library/Logs/gym/Runner-Runner.log
[11:36:34]: 
[11:36:34]: Looks like fastlane ran into a build/archive error with your project
[11:36:34]: It's hard to tell what's causing the error, so we wrote some guides on how
[11:36:34]: to troubleshoot build and signing issues: https://docs.fastlane.tools/codesigning/getting-started/
[11:36:34]: Before submitting an issue on GitHub, please follow the guide above and make
[11:36:34]: sure your project is set up correctly.
[11:36:34]: fastlane uses `xcodebuild` commands to generate your binary, you can see the
[11:36:34]: the full commands printed out in yellow in the above log.
[11:36:34]: Make sure to inspect the output above, as usually you'll find more error information there

```
</details>

<details>
  <summary>Runner-Runner.log</summary>

```
PhaseScriptExecution [CP]\ Embed\ Pods\ Frameworks /Users/jenkins/Library/Developer/Xcode/DerivedData/Runner-fsgvokzjdkqkrmbocrmrxbuzfayy/Build/Intermediates.noindex/ArchiveIntermediates/Runner/IntermediateBuildFilesPath/Runner.build/Release-iphoneos/Runner.build/Script-5B61BB44B89A7C0231C51373.sh (in target 'Runner' from project 'Runner')
    cd /Users/jenkins/dev/xxxxx/ios
    /bin/sh -c /Users/jenkins/Library/Developer/Xcode/DerivedData/Runner-fsgvokzjdkqkrmbocrmrxbuzfayy/Build/Intermediates.noindex/ArchiveIntermediates/Runner/IntermediateBuildFilesPath/Runner.build/Release-iphoneos/Runner.build/Script-5B61BB44B89A7C0231C51373.sh
mkdir -p /Users/jenkins/Library/Developer/Xcode/DerivedData/Runner-fsgvokzjdkqkrmbocrmrxbuzfayy/Build/Intermediates.noindex/ArchiveIntermediates/Runner/BuildProductsPath/Release-iphoneos/Runner.app/Frameworks
Source: /Users/jenkins/Library/Developer/Xcode/DerivedData/Runner-fsgvokzjdkqkrmbocrmrxbuzfayy/Build/Intermediates.noindex/ArchiveIntermediates/Runner/BuildProductsPath/Release-iphoneos/FMDB/FMDB.framework
Symlinked...
Linked: ../../../IntermediateBuildFilesPath/UninstalledProducts/iphoneos/FMDB.framework
rsync --delete -av --filter P .*.?????? --links --filter "- CVS/" --filter "- .svn/" --filter "- .git/" --filter "- .hg/" --filter "- Headers" --filter "- PrivateHeaders" --filter "- Modules" "../../../IntermediateBuildFilesPath/UninstalledProducts/iphoneos/FMDB.framework" "/Users/jenkins/Library/Developer/Xcode/DerivedData/Runner-fsgvokzjdkqkrmbocrmrxbuzfayy/Build/Intermediates.noindex/ArchiveIntermediates/Runner/InstallationBuildProductsLocation/Applications/Runner.app/Frameworks"
building file list ... rsync: link_stat "/Users/jenkins/dev/xxxxx/ios/../../../IntermediateBuildFilesPath/UninstalledProducts/iphoneos/FMDB.framework" failed: No such file or directory (2)
done

sent 29 bytes  received 20 bytes  98.00 bytes/sec
total size is 0  speedup is 0.00
rsync error: some files could not be transferred (code 23) at /AppleInternal/Library/BuildRoots/810eba08-405a-11ed-86e9-6af958a02716/Library/Caches/com.apple.xbs/Sources/rsync/rsync/main.c(996) [sender=2.6.9]
Command PhaseScriptExecution failed with a nonzero exit code
```
</details>

## Analysis

I traced the issue down to a single line in `Targets Support Files/Pods-Runner/Pods-Runner-frameworks.sh` (I added some debug outputs as seen in the previous log).

The script wants to copy the following framework:

```
/Users/jenkins/Library/Developer/Xcode/DerivedData/Runner-fsgvokzjdkqkrmbocrmrxbuzfayy/Build/Intermediates.noindex/ArchiveIntermediates/Runner/BuildProductsPath/Release-iphoneos/FMDB/FMDB.framework
```

which is a symlink:

```
/Users/jenkins/Library/Developer/Xcode/DerivedData/Runner-fsgvokzjdkqkrmbocrmrxbuzfayy/Build/Intermediates.noindex/ArchiveIntermediates/Runner/BuildProductsPath/Release-iphoneos/FMDB/FMDB.framework -> ../../../IntermediateBuildFilesPath/UninstalledProducts/iphoneos/FMDB.framework
```

When resolving the symlink in the script:

```
if [ -L "${source}" ]; then
  echo "Symlinked..."
  source="$(readlink "${source}")"
  echo "Linked: ${source}"
fi
```

it gets a relative path, and appends it to the flutterapp/ios directory, which causes the script to not find the directory.

A fix would be to call `readlink -f` to get an absolute link path:

```
if [ -L "${source}" ]; then
  echo "Symlinked..."
  source="$(readlink -f "${source}")"
  echo "Linked: ${source}"
fi
```

In this case the build succeeds.

**Comment by codeguesser** [comment]
works for me

**Comment by smiera** [comment]
this fix works, but if run `flutter clean ; flutter pub get ; flutter build ipa --release`, then file `Pods-Runner-frameworks.sh` re-created without `readlink -f`

**Comment by OlehMarchenko95** [comment]
Based on your analysis l've came up with following solution for.Cl:

1. I've copied script **Pods-Runner-frameworks.sh** into Runner directory
2. Edited this file as you've suggested
3. Changed **[CP] Embed Pods Frameworks** as follows:
```bash
# Uncomment when it will be fixed
# "${PODS_ROOT}/Target Support Files/Pods-Runner/Pods-Runner-frameworks.sh"
# For now using local script
"${PROJECT_DIR}/Runner/Pods-Runner-frameworks.sh"
```

PS: Not the best solution, but for now is ok.
PPS: Better would be to modify original script.

**Comment by virtualmarc** [comment]
> this fix works, but if run `flutter clean ; flutter pub get ; flutter build ipa --release`, then file `Pods-Runner-frameworks.sh` re-created without `readlink -f`

I worked around this by replacing the line via sed in my jenkins build script before calling fastlane.
Not the best solution but works for now.

**Comment by imdatsolak** [comment]
> Based on your analysis l've came up with following solution for.Cl:
> 
> 1. I've copied script **Pods-Runner-frameworks.sh** into Runner directory
> 2. Edited this file as you've suggested
> 3. Changed **[CP] Embed Pods Frameworks** as follows:
> 
> ```shell
> # Uncomment when it will be fixed
> # "${PODS_ROOT}/Target Support Files/Pods-Runner/Pods-Runner-frameworks.sh"
> # For now using local script
> "${PROJECT_DIR}/Runner/Pods-Runner-frameworks.sh"
> ```
> 
> PS: Not the best solution, but for now is ok. PPS: Better would be to modify original script.

Thank you very much. This solved it for me for the time being.

I'll do this manual process until the fix from flutter-team is available.

Tysm

**Comment by virtualmarc** [comment]
As mentioned by @hukusuke1007 in #123920  this seems to be a cocoapods issue https://github.com/CocoaPods/CocoaPods/issues/11808 that's going to fix our problem.

**Comment by utkuyaman** [comment]
that fix saved my day, thank you so much 🙏🏼

**Comment by curonny** [comment]
> works for me

Please, can you help me to solve.. i have the same error, but don't understand the solution

**Comment by codeguesser** [comment]
> > works for me
> 
> Please, can you help me to solve.. i have the same error, but don't understand the solution

<img width="994" alt="image" src="https://user-images.githubusercontent.com/1662117/229328423-77bb2b27-099b-48b1-9e99-60c3bf2d1d5a.png">
This may help you to solve the problem..If any questions quote to me

**Comment by anzbert** [comment]
this ```readlink -f``` did the trick for me. thanks so much.

**Comment by park012241** [comment]
I've added this script to `post_install` section in `Podfile` and it works on Xcode 14.3
I'm using cocoapods 1.12.0

```podfile
# Check out: https://github.com/flutter/flutter/issues/123852
# Workaround for Xcode 14.3:
system('sed -i \'\' \'44s/readlink/readlink -f/\' \'Pods/Target Support Files/Pods-Runner/Pods-Runner-frameworks.sh\'')
```

**Comment by serdarcida** [comment]
Hi, I have made a workaround below but it might save the day, flutter developers need to post an update for this issue, it seems like the issue is happening especially between stable flutter and latest xcode version

https://github.com/flutter/flutter/issues/123890

**Comment by sqxccdy** [comment]
thanks, work for me.

**Comment by darshankawar** [comment]
Thanks for the report everyone.
Per [this](https://github.com/flutter/flutter/issues/123852#issuecomment-1492855357) comment, since this seems to be related to cocoapods and seem to have been fixed at their end, https://github.com/CocoaPods/CocoaPods/pull/11828, until they release a new version of cocoapod, can everyone facing this issue in this thread try [this](https://github.com/CocoaPods/CocoaPods/pull/11828#issuecomment-1493406248) as mentioned and see if that helps to fix the issue at respective end ?

/cc @jmagman for visibility

**Comment by jmagman** [comment]
@darshankawar let's duplicate all of these https://github.com/CocoaPods/CocoaPods/issues/11808 reports to  https://github.com/flutter/flutter/issues/123890.

While this is a CocoaPods issue and has been fixed there, the Flutter team will investigate a workaround.  Please follow along in https://github.com/flutter/flutter/issues/123890 for updates.

**Comment by mukai154** [comment]
> > > works for me
> > 
> > 
> > Please, can you help me to solve.. i have the same error, but don't understand the solution
> 
> <img alt="image" width="994" src="https://user-images.githubusercontent.com/1662117/229328423-77bb2b27-099b-48b1-9e99-60c3bf2d1d5a.png">
> 
> This may help you to solve the problem..If any questions quote to me

THIS WORKS

THANK YOU!

**Comment by carloseduardovieira** [comment]
> ## Steps to Reproduce
> 1. Execute `flutter build ios`
> 2. Switch to ios subdirectory
> 3. Execute `pub install`
> 4. Run fastlane target containing:
> 
> ```
> build_app(
>   scheme: "Runner",
>   archive_path: "./build/Runner.xcarchive",
>   export_method: "app-store",
>   output_directory: "./build/Runner"
> )
> ```
> 
> **Expected results:** Build successful
> 
> **Actual results:** Build failed
> 
> ## Environment
> * Mac Mini M1 2020 with macOS Ventura 13.0.1
> * Xcode 14.3 (14E222b)
> * Flutter 3.7.9
> 
> ```
> Flutter 3.7.9 • channel stable • https://github.com/flutter/flutter.git
> Framework • revision 62bd79521d (16 hours ago) • 2023-03-30 10:59:36 -0700
> Engine • revision ec975089ac
> Tools • Dart 2.19.6 • DevTools 2.20.1
> ```
> 
> ## Build Logs
> Build Output
> Runner-Runner.log
> ## Analysis
> I traced the issue down to a single line in `Targets Support Files/Pods-Runner/Pods-Runner-frameworks.sh` (I added some debug outputs as seen in the previous log).
> 
> The script wants to copy the following framework:
> 
> ```
> /Users/jenkins/Library/Developer/Xcode/DerivedData/Runner-fsgvokzjdkqkrmbocrmrxbuzfayy/Build/Intermediates.noindex/ArchiveIntermediates/Runner/BuildProductsPath/Release-iphoneos/FMDB/FMDB.framework
> ```
> 
> which is a symlink:
> 
> ```
> /Users/jenkins/Library/Developer/Xcode/DerivedData/Runner-fsgvokzjdkqkrmbocrmrxbuzfayy/Build/Intermediates.noindex/ArchiveIntermediates/Runner/BuildProductsPath/Release-iphoneos/FMDB/FMDB.framework -> ../../../IntermediateBuildFilesPath/UninstalledProducts/iphoneos/FMDB.framework
> ```
> 
> When resolving the symlink in the script:
> 
> ```
> if [ -L "${source}" ]; then
>   echo "Symlinked..."
>   source="$(readlink "${source}")"
>   echo "Linked: ${source}"
> fi
> ```
> 
> it gets a relative path, and appends it to the flutterapp/ios directory, which causes the script to not find the directory.
> 
> A fix would be to call `readlink -f` to get an absolute link path:
> 
> ```
> if [ -L "${source}" ]; then
>   echo "Symlinked..."
>   source="$(readlink -f "${source}")"
>   echo "Linked: ${source}"
> fi
> ```
> 
> In this case the build succeeds.

I love you

**Comment by firosva** [comment]
> > > > works for me
> > > 
> > > 
> > > Please, can you help me to solve.. i have the same error, but don't understand the solution
> > 
> > 
> > <img alt="image" width="994" src="https://user-images.githubusercontent.com/1662117/229328423-77bb2b27-099b-48b1-9e99-60c3bf2d1d5a.png">
> > This may help you to solve the problem..If any questions quote to me
> 
> THIS WORKS
> 
> THANK YOU!

Thank you.. Its works for me.  :)

**Comment by jmagman** [comment]
This issue was duplicated to https://github.com/flutter/flutter/issues/123890.

> This fix is available on Flutter 3.10. Run `flutter upgrade` on the stable channel to get it. Please file any new issues you find. https://github.com/flutter/flutter/wiki/Hotfixes-to-the-Stable-Channel#3710-apr-05-2023 Thanks!
